corrected download.py

rnihesh opened the pull request:
Fixes #1 
- corrected download.py for better exception handling
- removed `-test` , name from the bucket var

vanga commented:
https://github.com/vanga/indian-supreme-court-judgments/actions/runs/16718819266/job/47412321869
there might be something wrong with the periodic update logic.. the logs here don't look right (metadata packaging when no pdfs were downloaded?)

rnihesh commented:
@vanga , hey , found the issue, and corrected it.

- when index file is being downloaded for the sake of `updated_at` key, its using the same `LOCAL_DIR` , so 2025 folder inside that, so parquet logic thinks that, It got new files, but really, we didn't get, 
- so, changed the path for latest date fetching via index file and gitignored it
